torch/onnx/utils.py

```diff
@@ -233,7 +233,11 @@ def _model_to_graph(model, args, f, verbose=False, training=False,
             raise RuntimeError('\'forward\' method must be a script method')
     else:
         graph, torch_out = _trace_and_get_graph_from_model(model, args, training)
-        params = list(_unique_state_dict(model).values())
+        params = list(_unique_state_dict(model, keep_vars=True).values())
+
+    input_and_param_names = [val.uniqueName() for val in graph.inputs()]
```

Comment:
Here we didn't use inputs provided by users to update our param_dict.

Reply:
That's right. Fixed.